async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -23,6 +23,7 @@ import org.reactivestreams.Publisher
 import _root_.io.netty.handler.codec.http.cookie.Cookie
 import org.asynchttpclient.uri.Uri
 import org.asynchttpclient.cookie.CookieStore
+import org.http4s.implicits._
```

Comment:
This is probably about the biggest breaking change that will come from this approach.  I'm okay with it, but we've recently had discussions about http4s' high import tax.  Also some hit to discoverability, but tooling has gotten a lot better since this was first proposed a long time ago.

If we do this, and remove some imports from the server and client DSLs, it's a winning tradeoff.